pkg/client/restclient/urlprovider.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restclient
+
+import (
+	"net/url"
+	"sync"
+)
+
+// A URLProvider allows to use multiple URLs.
+type URLProvider interface {
+	// Get returns currently selected URL.
+	Get() *url.URL
+	// Next selects any other URL, it is upto implementation to provide any
+	// health checking/load balancing, it might be as simple as an iterator.
+	Next() *url.URL
+	// Total returns number of available URLs.
+	Total() int
```

Comment:
maybe i shouldn't be adding this interface at all and just use RoundRobinProvider everywhere